src/Shared/Http2cat/SslDuplexPipe.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Security;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+
+namespace Microsoft.AspNetCore.Http2Cat
+{
+    internal class SslDuplexPipe : DuplexPipeStreamAdapter<SslStream>
+    {
+        public SslDuplexPipe(IDuplexPipe transport, StreamPipeReaderOptions readerOptions, StreamPipeWriterOptions writerOptions)
```

Comment:
Why was it necessary to duplicate this? I think we need to figure out a better story for all of our DuplexPipes (put all of them in a shared code place).

Reply:
Nvrm, I see you moved a lot to ServerInfrastructure folder. Can this live there as well?

Reply:
Moved.